AsyncSwift/Observed/ProfileView/ProfileViewObserved.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  ProfileView+Observed.swift
+//  AsyncSwift
+//
+//  Created by Kim Insub on 2022/10/16.
+//
+
+import CodeScanner
+import CoreImage.CIFilterBuiltins
+import Combine
+import UIKit
+
+
+// TODO
+// 1. hasRegisteredProfile -> keyChain 으로 옮기기
+
+final class ProfileViewObserved: ObservableObject {
+    @Published var hasRegisteredProfile = false
+    @Published var isLoading = true
+    @Published var isShowingFriends = false
+    @Published var isShowingEdit = false
+    @Published var isShowingScanner = false
+    @Published var isShowingUserDetail = false
+    @Published var user: User = User(
+        id: "",
+        name: "",
+        nickname: "",
+        role: "",
+        description: "",
+        linkedInURL: "",
+        profileURL: "",
+        friends: []
+    )
+
+    @Published var scannedFriend: User = User(
+        id: "",
+        name: "",
+        nickname: "",
+        role: "",
+        description: "",
+        linkedInURL: "",
+        profileURL: "",
+        friends: []
+    )
+
+    var userID: String? = nil {
+        didSet {
+            let _ = KeyChain.shared.addItem(key: "userID", pwd: userID ?? "")
+        }
+    }
+
+    init() {
+        let userid = KeyChain.shared.getItem(key: "userID")
+        guard userid != nil else { return }
+        self.hasRegisteredProfile = true
+        self.userID = userid as? String
+    }
+
+    func onAppear() {
+        if hasRegisteredProfile {
+            getUserByID()
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) { [weak self] in
+                guard let self = self else { return }
+                self.isLoading = false
+            }
+        }
+    }
+
+    func didTapXButton() {
```

Comment:
Close button이 더 적절해보여요 ~
```suggestion
    func didTapCloseButton() {
```

Reply:
[refactor : Naming -> didTapCloseButton](https://github.com/Async-Swift/AsyncSwift/pull/60/commits/02e7faac7b4dff860a14aabd51be027e494addb5)

수정하였습니다